packages/odyssey-react-mui/src/labs/GroupPicker.tsx

```diff
@@ -0,0 +1,247 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type { AutocompleteGetTagProps } from "@mui/material/useAutocomplete";
+
+import {
+  Autocomplete as MuiAutocomplete,
+  AutocompleteProps as MuiAutocompleteProps,
+  Box,
+  Chip as MuiChip,
+  InputBase,
+  Avatar as MuiAvatar,
+} from "@mui/material";
+import { HTMLAttributes, memo, useCallback } from "react";
+
+import { AutocompleteProps } from "../Autocomplete";
+import { Field } from "../Field";
+import { Subordinate } from "../Typography";
+import { useOdysseyDesignTokens } from "../OdysseyDesignTokensContext";
+
+import { UserIcon, GridIcon, GroupIcon } from "../icons.generated";
+
+export type GroupPickerOptionType = {
+  id: string;
+  name: string;
+  description: string;
+  logo?: string;
+  usersCount?: number;
+  appsCount?: number;
+  groupPushMappingsCount?: number;
+};
+
+export type GroupPickerProps<
+  GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+> = {
+  /**
+   * Hide selected options
+   */
+  filterSelectedOptions?: MuiAutocompleteProps<
+    GroupPickerOptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["filterSelectedOptions"];
+} & AutocompleteProps<
+  GroupPickerOptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>;
+
+const GroupPicker = <
+  optionType extends GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+>({
+  hasMultipleChoices,
+  isCustomValueAllowed,
+  isDisabled,
+  isLoading,
+  isOptional = false,
+  isReadOnly,
+  hint,
+  label,
+  onChange,
+  onInputChange,
+  options,
+  value,
+  filterSelectedOptions,
+  testId,
+}: GroupPickerProps<optionType, HasMultipleChoices, IsCustomValueAllowed>) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const isOptionEqualToValue = useCallback(
+    (sourceValue: optionType, targetValue: optionType) => {
+      return sourceValue.id === targetValue.id;
+    },
+    []
+  );
+
+  const getOptionLabel = useCallback((option: string | optionType) => {
+    return typeof option === "string" ? option : option.name;
+  }, []);
+
+  const renderOption = useCallback(
+    (props: HTMLAttributes<HTMLLIElement>, option: optionType) => {
+      return (
+        <li {...props} key={option.id}>
+          <Box
+            sx={{
+              display: "flex",
+              flexDirection: "row",
+              alignItems: "top",
+            }}
+          >
+            <Box sx={{ paddingRight: odysseyDesignTokens.Spacing2 }}>
+              <MuiAvatar
+                sx={{
+                  backgroundColor: "transparent",
+                  width: 24,
+                  height: 24,
```

Comment:
Can we move this as a static variable at the top of a function? I don't like magic numbers if we can avoid them. Even still, I didn't know these were the image's size until you told me.

Something like `const avatarImageSize = 24` above the function.